conans/client/generators/cmake_common.py

```diff
@@ -397,6 +397,10 @@ def generate_targets_section(dependencies):
 endfunction()
 
 function(conan_check_compiler)
+    if(CONAN_DISABLE_CHECK_COMPILER)
+        conan_message(STATUS "WARN: Disabled conan compiler checks")
+        return()
+    endif()
     if(NOT DEFINED CMAKE_CXX_COMPILER_ID)
```

Comment:
I suppose we also have to check `ENABLED_LANGUAGES` property
(to obtain a list use `get_property(languages GLOBAL PROPERTY ENABLED_LANGUAGES)`)
and execute checks only if it has `C` or `CXX` in the list (potentially, also `Objective-C` and `Objective-C++`?), otherwise skip (no language at all / Fortran / D / Rust / Go / C# / Whatever else)